vars/runJenkinsTests.groovy

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env groovy
+
+def call(Map options = [:]) {
+    def jenkinsAdminPassword = sh(returnStdout: true, script: options.sshCommand + " sudo cat /var/lib/jenkins/secrets/initialAdminPassword").trim()
+    def version = null
+    try {
+        // NOTE: The password will be printed out in the logs, but that's fine since you still have to ssh onto the VM to use it
+        version = sh(returnStdout: true, script: 'curl --silent "' + options.utilsLocation + 'jenkins/run-cli-command.sh" | sudo bash -s -- -c "version" -ju "admin" -jp ' + jenkinsAdminPassword).trim()
+        echo "Jenkins version: " + version
+    } catch (e) {
+    }
+
+    if (!version || version == "") {
+        error("Failed to get Jenkins version.")
+    }
+
+    //verify that the Azure landing page works
+    def remoteURL = options.sshCommand.split('@')[1].split(' ')[0]
+    print remoteURL;
+
+    def compare_script_path = 'scripts/compare-web-pages.sh'
+    sh 'sudo chmod +x ' + compare_script_path
+
+    //we're using the login page, because http://localhost doest javascript redirect to login
+    diff_result = sh(returnStatus: true, script: compare_script_path + " -f http://localhost:8080/login -s http://" + remoteURL + "/login")
+    if (diff_result != 1) {
+        error("The public landing page is identical with the local page!")
```

Comment:
When I proposed searching for the word "Azure", it was as a simpler replacement for the diff-ing. Something like this:
```groovy
remote_page = sh(returnStdout: true, script: "curl -s -L http://" + remoteURL + "/login")
if (remote_page.contains("<title>Jenkins</title>")) {
  error("The login page is accessible over the public IP address, possibly exposing the user to Man in the Middle Attacks!")
} else if (!remote_page.contains("<title>Jenkins On Azure</title>")) {
  error("The Azure landing page was not displayed over the public IP address!")
}
```

You could do something similar for the local page, but I'm not sure that provides much value:
```groovy
local_page = sh(returnStdout: true, script: "curl -s -L http://localhost:8080/login")
if (!local_page.contains("<title>Jenkins</title>")) {
  error("The login page was not displayed over localhost!")
}
```